File: miniprogram/types/submission.ts
export interface PhotoInfo {
  url: string;
  file_size: number;
  width: number;
  height: number;
  deleted: boolean;
}

export interface AIEvaluationBreakdown {
  clarity?: number;
  lighting?: number;
  angle?: number;
  background?: number;
  expression?: number;
  composition?: number;
}

export interface AIEvaluation {
  status: '' | 'pending' | 'success' | 'failed';
  score: number;
  issues: string[];
  suggestions: string[];
  breakdown: AIEvaluationBreakdown;
  evaluated_at?: string;
  error?: string;
}

export interface UserInfo {
  nick_name: string;
  avatar_url: string;
}

export interface Submission {
  id: string;
  task_id: string;
  user_id: string;
  user_info: UserInfo;
  custom_data: Record<string, string | string[]>;
  photo: PhotoInfo;
  ai_evaluation: AIEvaluation;
  status: 'draft' | 'submitted' | 'rejected';
  created_at: string;
  updated_at: string;
}

export interface SubmissionListResponse {
  list: Submission[];
  total: number;
  has_more: boolean;
}

export interface SubmitPhotoParams {
  task_id: string;
  photo: {
    url: string;
    file_size?: number;
    width?: number;
    height?: number;
  };
  custom_data?: Record<string, any>;
}

export interface SubmissionAnalysisResult {
  model: string;
  score: number;
  breakdown: AIEvaluationBreakdown;
  issues: string[];
  suggestions: string[];
}

File: backend/internal/data/submission.go
package data

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type PhotoInfo struct {
	URL      string    `bson:"url" json:"url"`
	FileSize int64     `bson:"file_size" json:"file_size"`
	Width    int       `bson:"width" json:"width"`
	Height   int       `bson:"height" json:"height"`
	Deleted  bool      `bson:"deleted" json:"deleted"`
}

type AIEvaluation struct {
	Status      string            `bson:"status" json:"status"`
	Score       int               `bson:"score" json:"score"`
	Issues      []string          `bson:"issues" json:"issues"`
	Suggestions []string          `bson:"suggestions" json:"suggestions"`
	Breakdown   map[string]int    `bson:"breakdown" json:"breakdown"`
	EvaluatedAt time.Time         `bson:"evaluated_at,omitempty" json:"evaluated_at,omitempty"`
}

type UserInfo struct {
	NickName  string `bson:"nick_name" json:"nick_name"`
	AvatarURL string `bson:"avatar_url" json:"avatar_url"`
}

type Submission struct {
	ID           primitive.ObjectID     `bson:"_id,omitempty" json:"id"`
	TaskID       primitive.ObjectID     `bson:"task_id" json:"task_id"`
	UserID       string                 `bson:"user_id" json:"user_id"`
	VerificationCode string             `bson:"-" json:"verification_code,omitempty"`
	UserInfo     UserInfo               `bson:"user_info" json:"user_info"`
	CustomData   map[string]interface{} `bson:"custom_data" json:"custom_data"`
	Photo        PhotoInfo              `bson:"photo" json:"photo"`
	AIEvaluation AIEvaluation           `bson:"ai_evaluation" json:"ai_evaluation"`
	Status       string                 `bson:"status" json:"status"`
	CreatedAt    time.Time              `bson:"created_at" json:"created_at"`
	UpdatedAt    time.Time              `bson:"updated_at" json:"updated_at"`
}
